src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Statistics/BoundaryDataControllerTests.cs

```diff
@@ -0,0 +1,106 @@
+public class BoundaryDataControllerTests
+{
+    private readonly Mock<IBoundaryDataService> _boundaryDataService;
+    private readonly Mock<IBoundaryLevelService> _boundaryLevelService;
+    private readonly BoundaryDataController _sut;
+
+    public BoundaryDataControllerTests()
+    {
+        _boundaryDataService = new Mock<IBoundaryDataService>(MockBehavior.Strict);
+        _boundaryLevelService = new Mock<IBoundaryLevelService>(MockBehavior.Strict);
+
+        _sut = new(_boundaryDataService.Object, _boundaryLevelService.Object);
+    }
+
+    [Fact]
+    public async Task GetBoundaryLevels_ReturnsOk()
+    {
+        // Arrange
+        var boundaryLevels = new List<BoundaryLevelViewModel>
+        {
+            new() {Id = 1, Level = "School", Label = "Test", Published = DateTime.UtcNow},
+        };
+
+        _boundaryLevelService
+            .Setup(bls => bls.Get())
+            .ReturnsAsync(boundaryLevels);
+
+        // Act
+        var result = await _sut.GetBoundaryLevels();
+
+        // Assert
+        Assert.IsAssignableFrom<OkObjectResult>(result.Result);
+    }
+
+    [Fact]
+    public async Task GetBoundaryLevel_MissingId_ReturnsBadRequest()
+    {
+        // Act
+        var result = await _sut.GetBoundaryLevel(0);
+
+        // Assert
+        Assert.IsAssignableFrom<BadRequestObjectResult>(result.Result);
+    }
+
+    [Fact]
+    public async Task GetBoundaryLevel_ReturnsOk()
+    {
+        // Arrange
+        var boundaryLevel = new BoundaryLevelViewModel { Id = 1, Level = "School", Label = "Test", Published = DateTime.UtcNow };
+
+        _boundaryLevelService
+            .Setup(bls => bls.Get(1))
+            .ReturnsAsync(boundaryLevel);
+
+        // Act
+        var result = await _sut.GetBoundaryLevel(1);
+
+        // Assert
+        Assert.IsAssignableFrom<OkObjectResult>(result.Result);
+    }
+
+    [Fact]
+    public async Task UpdateBoundaryLevel_ReturnsOk()
+    {
+        // Arrange
+        _boundaryLevelService
+            .Setup(bls => bls.UpdateLabel(1, "Test (updated)"))
+            .Returns(Task.CompletedTask);
+
+        // Act
+        var result = await _sut.UpdateBoundaryLevel(new() { Id = 1, Label = "Test (updated)" });
+
+        // Assert
+        Assert.IsAssignableFrom<NoContentResult>(result);
+    }
+
+    [Fact]
+    public async Task UploadBoundaryFile_InvalidFileExtension_ReturnsBadRequest()
+    {
+        // Arrange
+        var geoJsonFile = MockFormTestUtils.CreateFormFileMock("boundaries.json", "application/octet-stream").Object;
+
+        // Act
+        var response = await _sut.UploadBoundaryFile(
+            GeographicLevel.School,
+            "Boundary Data 1",
+            geoJsonFile,
+            DateTime.UtcNow);
+
+        // Assert
+        Assert.IsAssignableFrom<BadRequestObjectResult>(response);
+        Assert.Equal("Invalid file format. This function only accepts GeoJSON files", (response as BadRequestObjectResult).Value);
+    }
```

Comment:
See comment left in `BoundaryLevelServiceTests.cs` about switching to integration tests.